### PR TITLE
Enable bin set selection when evaluating gaseous absorption in CKD mode

### PR DESCRIPTION
**Pull request by leroyvn**
# Description

This PR enables the automatic selection of CKD absorption data based on the active bin set. Changes are as follows:

- [x] `CKDSpectralContext` has a new optional `bin_set` field, which can be set by the `MeasureSpectralConfig` which emits it.
- [x] The following methods are added a `bin_set` parameter used to select a relevant CKD dataset.
  - `RadProfile.eval_sigma_a_ckd()`
  - `RadProfile.eval_dataset_ckd()`
  - `US76ApproxRadProfile.eval_sigma_t_ckd()`
  - `US76ApproxRadProfile.eval_albedo_ckd()`
  - `AFGL1986RadProfile.eval_sigma_t_ckd()`
  - `AFGL1986RadProfile.eval_albedo_ckd()`
- [x] `RadProfile.eval_sigma_a()` and `RadProfile.eval_dataset()` use the `bin_set` field of its `spectral_ctx` parameter to call `RadProfile.eval_sigma_a_ckd()` and `RadProfile.eval_dataset_ckd()` appropriately, respectively.
- [x] Changed back the tolerance of `test_onedim_vkd_vs_mono` from 10 % to 5 % ; the original tolerance was 5 %, it was changed to 10 % when the "10nm" bin set was replaced by the "1nm" bin set but this setting had no action hence the existence of this PR.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by leroyvn** [comment]
@nollety the floor is yours!

**Comment by nollety** [comment]
@leroyvn Thanks, I'm on it!

File: src/eradiate/contexts.py [hunk 1]
```diff
@@ -224,6 +224,19 @@ class CKDSpectralContext(SpectralContext):
...
+        init_type=":class:`.Bindex` or tuple or dict, optional",
+    )
+
+    bin_set: t.Optional[BinSet] = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(BinSet.convert),
+            validator=attr.validators.optional(attr.validators.instance_of(BinSet)),
+        ),
+        doc="If relevant, the bin set from which ``bindex`` originates.",
```
Comment (nollety) [inline src/eradiate/contexts.py:236]: Why wouldn't it be relevant? I thought a bindex was ambiguous without the reference to the bin set.

Comment (leroyvn) [reply src/eradiate/contexts.py:236]: Unless you have to select a bin set-dependent data file, you don't care about the bin set ID. Most components can actually evaluate optical properties in CKD modes without information about the bin set ID, gaseous absorption data is a notable exception. So I see no reason to always require bin set information when creating a `CKDSpectralContext` object.

Comment (nollety) [reply src/eradiate/contexts.py:236]: Ah that's right, I forgot about that

File: src/eradiate/radprops/rad_profile.py [hunk 2]
```diff
@@ -367,7 +369,7 @@ def eval_sigma_a_mono(self, w: pint.Quantity) -> pint.Quantity:
         """
         raise NotImplementedError
 
-    def eval_sigma_a_ckd(self, *bindexes: Bindex) -> pint.Quantity:
+    def eval_sigma_a_ckd(self, *bindexes: Bindex, bin_set: str) -> pint.Quantity:
```
Comment (leroyvn) [inline src/eradiate/radprops/rad_profile.py:372]: I overlooked this: for clarity, I rather suggest to name this arg `bin_set_id`.

**Review by leroyvn — COMMENTED** [review]
Hi @nollety, looks good! I'll request one change only (bad anticipation from me).

File: src/eradiate/radprops/rad_profile.py (hunk 2 above)
Comment (nollety) [reply src/eradiate/radprops/rad_profile.py:372]: Yeah, that makes sense, I'll do that

**Comment by nollety** [comment]
@leroyvn Running the slow tests to confirm that it did not break anything

**Comment by nollety** [comment]
This should be ready in ~30 minutes

**Comment by nollety** [comment]
All tests passed! 🎆

**Comment by leroyvn** [comment]
Ok, I'll clean up the commit history and merge. Thanks @nollety!

**Pull request merged**